vector/src/main/java/fr/gouv/tchap/adapters/TchapContactAdapter.java

```diff
@@ -395,10 +395,26 @@ private void populateViews(final ParticipantAdapterItem participant, final int p
                 vContactName.setTypeface(null, Typeface.BOLD);
             }
 
+            // display the participant's avatar
             participant.displayAvatar(mSession, vContactAvatar);
 
+            // display the participant's name
             vContactName.setText(DinsicUtils.getNameFromDisplayName(participant.mDisplayName));
-            vContactDomain.setText(DinsicUtils.getDomainFromDisplayName(participant.mDisplayName));
+
+            // display the participant's domain
+            String domainName = DinsicUtils.getDomainFromDisplayName(participant.mDisplayName);
+
+            if (null == domainName || domainName.isEmpty()) {
```

Comment:
First the domain is relevant only for Tchap users.
We should handle differently vContactName and vContactDomain according to the test: "participant.isMatrixUser()".

Before accessing to the contact emails, you must check (null != participant.mContact) and (!participant.mContact.getEmails().isEmpty()) even if we know that these conditions are satisfied presently, in order to be robust to potential futur changes.

Please add a comment that we will extract the domain of this tchap user from the his email.